recipe-client-addon/test/browser/browser_RecipeRunner.js

```diff
@@ -152,17 +154,45 @@ add_task(async function testStart() {
     ["extensions.shield-recipe-client.experiments.lazy_classify", false],
   ]});
   ok(!getStub.called, "getClientClassification hasn't been called");
-  await RecipeRunner.start();
-  ok(getStub.called, "getClientClassfication was called eagerly");
+  await RecipeRunner.run();
+  ok(getStub.called, "getClientClassification was called eagerly");
 
   // When the experiment pref is true, do not eagerly call getClientClassification.
   await SpecialPowers.pushPrefEnv({set: [
     ["extensions.shield-recipe-client.experiments.lazy_classify", true],
   ]});
   getStub.reset();
   ok(!getStub.called, "getClientClassification hasn't been called");
-  await RecipeRunner.start();
-  ok(!getStub.called, "getClientClassfication was not called eagerly");
+  await RecipeRunner.run();
+  ok(!getStub.called, "getClientClassification was not called eagerly");
 
   getStub.restore();
 });
+
+add_task(async function testStartup() {
+  const runStub = sinon.stub(RecipeRunner, "run");
+  const addCleanupHandlerStub = sinon.stub(CleanupManager, "addCleanupHandler");
+  const updateRunIntervalStub = sinon.stub(RecipeRunner, "updateRunInterval");
+
+  // in dev mode
+  await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", true]]});
+  RecipeRunner.init();
+  ok(runStub.called, "RecipeRunner.run is called immediately when in dev mode");
+  ok(addCleanupHandlerStub.called, "A cleanup function is registered when in dev mode");
+  ok(updateRunIntervalStub.called, "A timer is registered when in dev mode");
+
+  runStub.reset();
+  addCleanupHandlerStub.reset();
+  updateRunIntervalStub.reset();
+
+  // not in dev mode
+  await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", false]]});
+  RecipeRunner.init();
+  ok(!runStub.called, "RecipeRunner.run is not called immediately when not in dev mode");
+  ok(addCleanupHandlerStub.called, "A cleanup function is registered when not in dev mode");
+  ok(updateRunIntervalStub.called, "A timer is registered when not in dev mode");
+
+  runStub.restore();
+  addCleanupHandlerStub.restore();
+  updateRunIntervalStub.restore();
+});
```

Comment:
Have you tested manually that the timer is working the way you want, that there aren't unexpected changes to other timers, that unregister works etc? I think changing the system clock in a VM would be enough, or maybe just fiddling with the pref.

I can't imagine why there would be a problem from looking at the timer manager code and it has tests, but it's probably something to confirm, especially if we're going to uplift this so it won't spend a lot of time on Nightly.

This shouldn't block merging the code, I think it's worth at least manually double-checking.